File: python/sorting/merge_sort_test.py
from .sort import SortType
from .merge_sort import MergeSort
from .perf_measurer import PerfMeasurer


def test_merge_sort():
    arr = [76, 34, 67, 32, 78, 45]
    sorter = MergeSort()
    sorter.sort(arr)

    expected_asc = [32, 34, 45, 67, 76, 78]
    assert arr == expected_asc

    sorter.sort(arr, SortType.desc)
    expected_desc = [78, 76, 67, 45, 34, 32]
    assert arr == expected_desc

    perf_measurer = PerfMeasurer(sorter)
    print(perf_measurer.reportPerf())

File: typescript/sorting/merge_sort.test.ts
import { assertEquals } from "https://deno.land/std@0.99.0/testing/asserts.ts";

import MergeSort from "./merge_sort.ts";
import { SORT_TYPE } from "./sort.ts";
import PerfMeasurer from "./perf_measurer.ts";

Deno.test("Merge sort", () => {
  const arr: number[] = [76, 34, 67, 32, 78, 45];
  const sorter: MergeSort = new MergeSort();

  const sortedAsc: number[] = sorter.sort(arr, SORT_TYPE.asc);
  const expectedAsc: number[] = [32, 34, 45, 67, 76, 78];
  assertEquals(sortedAsc, expectedAsc);

  const sortedDesc: number[] = sorter.sort(arr, SORT_TYPE.desc);
  const expectedDesc: number[] = [78, 76, 67, 45, 34, 32];
  assertEquals(sortedDesc, expectedDesc);

  const perfMeasurer = new PerfMeasurer(sorter);
  console.log(perfMeasurer.reportPerf());
});
